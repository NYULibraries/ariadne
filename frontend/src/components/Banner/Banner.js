import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { getInstitutionQueryParameter } from '../../aux/helpers';
import { bannerInstitutionInfo } from '../../aux/institutionConstants';

const Banner = () => {
  const institutionName = getInstitutionQueryParameter();
  const { logo, link, imgClass } = bannerInstitutionInfo[institutionName] || bannerInstitutionInfo.nyu;
  return (
    <Navbar className="color-nav" expand="lg">
      <Container>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <Nav.Link href={link}>
              <img className={imgClass} src={logo} alt="NYU Libraries logo" width="220" height="60" />
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Banner;
