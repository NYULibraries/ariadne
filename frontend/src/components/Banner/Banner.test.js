import { render, screen, waitFor } from '@testing-library/react';

import Banner from './Banner';

test('renders the Banner component', () => {
  render(<Banner />);
});

test('renders the NYU Libraries logo', async () => {
  render(<Banner />);
  const linkElement = await waitFor(() => screen.getByAltText(/NYU Libraries logo/i));
  expect(linkElement).toBeInTheDocument();
});

test('renders correctly', () => {
  const { asFragment } = render(<Banner />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders the correct NYUAD logo and link based on institution query parameter', async () => {
  window.history.pushState({}, null, '/?institution=NYUAD');
  render(<Banner />);
  const linkElement = await waitFor(() => screen.getByAltText(/NYU Libraries logo/i).closest('a'));
  expect(linkElement).toHaveAttribute('href', 'https://nyuad.nyu.edu/en/library.html');
  const imgElement = linkElement.querySelector('img');
  expect(imgElement).toHaveAttribute('src', `/images/abudhabi-logo-color.svg`);
});

test('renders the correct NYUSH logo and link based on institution query parameter', async () => {
  window.history.pushState({}, null, '/?institution=NYUSH');
  render(<Banner />);
  const linkElement = await waitFor(() => screen.getByAltText(/NYU Libraries logo/i).closest('a'));
  expect(linkElement).toHaveAttribute('href', 'https://shanghai.nyu.edu/academics/library');
  const imgElement = linkElement.querySelector('img');
  expect(imgElement).toHaveAttribute('src', `/images/shanghai-logo-color.svg`);
});

test('changes the background of the logo correctly when institution is NYUSH or NYUAD', async () => {
  const institution = 'NYUAD';
  render(<Banner />, {
    route: `?institution=${institution}`,
  });
  const linkElement = await waitFor(() => screen.getByAltText(/NYU Libraries logo/i));
  expect(linkElement).toHaveClass('image white-bg');
});
