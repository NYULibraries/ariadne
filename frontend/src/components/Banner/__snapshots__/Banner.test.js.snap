// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="color-nav navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="http://library.nyu.edu"
            href="http://library.nyu.edu"
          >
            <img
              alt="NYU Libraries logo"
              class="image"
              height="30"
              src="https://cdn.library.nyu.edu/images/nyulibraries-logo.svg"
              width="220"
            />
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
